### PR TITLE
Use different queue for version check timeouts

### PR DESCRIPTION
**Pull request by Sebastian-Hojas**
Using `first()` on the same queue causes the timeout
event to never be executed.

See https://github.com/Carthage/Carthage/issues/2544

### CODE REVIEW

**Comment by blender** [comment]
Wonderful! Thanks for the contribution! 

Do you think we can add a quick test for this?

**Comment by Sebastian-Hojas** [comment]
In order to make `remoteVersion()` testable, I assume we would need to mock the http request. Can you pinpoint me to some tests where you have already done this? Or did you have some other test strategy in mind?

**Comment by blender** [comment]
There are ideas of course, if you have better ones feel free to suggest.  I am also fine without tests for something till small but it would be a nice to have.

**Comment by Sebastian-Hojas** [comment]
Yeah, I can definitely see the value in this and would support said change. I can also think a bit about it, but I would suggest to not piggy-back the introduction of a mocking web server for tests on this tiny PR in particular. :)

**Comment by blender** [comment]
Maybe introduce another "firable" event that will never get fired in tests?

**Comment by blender** [comment]
A mock `Repository(owner: "Carthage", name: "Carthage").releases, perPage: 2)` ?

**Comment by Sebastian-Hojas** [comment]
You can't mock `Client` unfortunately as it's `final`, but we could mock the producer itself. Something like `public func remoteVersion(_ remoteVersionProducer: SignalProducer<_, Client.Error>) -> SemanticVersion? `

As far as I understand that, we would also need to move the public function from `BuildVersion.swift` to `CarthageKit`. Is my assumption correct?
I am wondering whether that semantically makes sense.

**Comment by Sebastian-Hojas** [comment]
Renamed to `RemoteVersion`, moved mockable `remoteVersion` to CarthageKit and added simple tests. Waiting for feedback now.

File: Source/CarthageKit/RemoteVersion.swift [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import Foundation
+import ReactiveSwift
+import ReactiveTask
+import Result
+import Tentacle
+
+/// Synchronously returns the newest Carthage release version,
```
Comment (blender) [inline Source/CarthageKit/RemoteVersion.swift]: Is the doc mentioning "Carthage" still relevant?

File: Source/carthage/RemoteVersion.swift [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import Foundation
+import CarthageKit
+import ReactiveSwift
+import ReactiveTask
+import Result
+import Tentacle
+
+/// The latest online version as a SemanticVersion object.
+public func remoteVersion() -> SemanticVersion? {
+	let remoteVersionProducer = Client(.dotCom, urlSession: URLSession.proxiedSession)
```
Comment (blender) [inline Source/carthage/RemoteVersion.swift]: awesome

File: Tests/CarthageKitTests/RemoteVersionSpec.swift [hunk 3]
```diff
@@ -0,0 +1,28 @@
+import Foundation
+import Nimble
+import Quick
+import ReactiveSwift
+import Tentacle
+
+@testable import CarthageKit
+
+class RemoteVersionSpec: QuickSpec {
```
Comment (blender) [inline Tests/CarthageKitTests/RemoteVersionSpec.swift]: great! Now it shall never break again!

**Comment by blender** [comment]
Just one question on the docs, then this is good to merge

File: Source/CarthageKit/RemoteVersion.swift (hunk 1 above)
Comment (Sebastian-Hojas) [reply Source/CarthageKit/RemoteVersion.swift]: Yeah, very good point. I updated the docs.

**Comment by Sebastian-Hojas** [comment]
Everything should be resolved now. Thanks for the feedback and the guidance - I am happy we got that issue resolved so easily!

**Comment by blender** [comment]
Awesome. Well done! Thanks for the contribution!

**Pull request merged**